### PR TITLE
Create docker-compose with Apache Atlas

### PR DESCRIPTION
**Pull request by bolkedebruin**
This is WIP as it seems the amundsen images are not publicly available?

(hence it references some 3rd party at the moment, but those builds are incorrect)

cc: @feng-tao @verdan

### CODE REVIEW

**Comment by feng-tao** [comment]
@verdan @bolkedebruin Once we have full integration with Apache Atlas,  it would be great to include a installation doc with sample start script like (https://github.com/lyft/amundsendatabuilder/blob/master/example/scripts/sample_data_loader.py) to bootstrap atlas with dummy data :)

File: docker-amundsen-atlas.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      ports:
+        - 5002:5000
+      networks:
+        - amundsennet
+      environment:
+        - CREDENTIALS_PROXY_USER=admin
+        - CREDENTIALS_PROXY_PASSWORD=admin
+        - PROXY_HOST=atlas
+        - PROXY_PORT=21000
+        - PROXY_CLIENT=ATLAS
```
Comment (verdan) [inline docker-amundsen-atlas.yml]: this should be `metadata_service.proxy.atlas_proxy.AtlasProxy` i.e., the class of the Proxy Client.

Comment (bolkedebruin) [reply docker-amundsen-atlas.yml]: I use the lookup table, so this can be either NEO4J or ATLAS at the moment (capitalization matters)

**Comment by bolkedebruin** [comment]
This should now work @verdan @feng-tao it includes example data but thereseems to be a glitch there (some columns are displayed as null)

**Comment by jornh** [comment]
@bolkedebruin I took it for a spin. Made sure I pulled and rebuilt latest metadata repo code with:
```
commit bbeeb59e6f576cd9eb5a6ab4edbae8b5ea379d20
Author: bolkedebruin <bolkedebruin@users.noreply.github.com>
Date:   Sun Apr 28 07:01:09 2019 +0200

    Describe and make easier Apache Atlas integration (#30)

    Updates the config to take PROXY_CLIENT from the environment.
```

When I look at the container log for the metadata service this is part of what I see:
```
2019-04-28T21:48:49+0000.765 [DEBUG] connectionpool._new_conn:205 (1:Thread-3) - Starting new HTTP connection (1): atlas:21000
2019-04-28T21:48:50+0000.166 [DEBUG] connectionpool._make_request:393 (1:Thread-3) - http://atlas:21000 "GET /api/atlas/v2/search/basic?typeName=Table&excludeDeletedEntities=true HTTP/1.1" 400 None
2019-04-28T21:48:50+0000.169 [ERROR] atlas_proxy.get_popular_tables:292 (1:Thread-3) - Please make sure you have assigned the appropriate self.TABLE_ENTITY entity to your atlas tables. HTTP request failed for GET http://atlas:21000/api/atlas/v2/search/basic?typeName=Table&excludeDeletedEntities=true: Bad request 400: {"errorCode":"ATLAS-400-00-00A","errorMessage":"Table: Unknown/invalid typename"}
``` 
When I look in the Atlas web user interface then among other types I see `Asset`, `Dataset` and `rokku_clent` so it seems the compose files `ATLAS_PROVISION_EXAMPLES=true` has an effect - but there are no `Table` type in my Atlas instance. And also that request are routed right from Amundsen frontend through the metadata service to .

I hope this gives you a clue about what could be going wrong. Let me know what else you need me to do to help.

**Comment by bolkedebruin** [comment]
Hi @jornh sorry forgot to mention it is dependent on a new release of the atlas docker build. The examples are actually not provisioned in yours.

**Comment by jornh** [comment]
@bolkedebruin: 🎉 I just spotted a new 0.0.9 image, so with:

```
  atlas:
      image: wbaa/rokku-dev-apache-atlas:0.0.9
```
(maybe worth pinning to a known good - for reproducibility when people report stuff?)

I now get:

```
atlas_amundsen      |
atlas_amundsen      | Creating sample types:
atlas_amundsen      | Created type [DB]
atlas_amundsen      | Created type [Table]
atlas_amundsen      | Created type [StorageDesc]
atlas_amundsen      | Created type [Column]
atlas_amundsen      | Created type [LoadProcess]
atlas_amundsen      | Created type [View]
atlas_amundsen      | Created type [JdbcAccess]
atlas_amundsen      | Created type [ETL]
atlas_amundsen      | Created type [Metric]
atlas_amundsen      | Created type [PII]
atlas_amundsen      | Created type [Fact]
atlas_amundsen      | Created type [Dimension]
atlas_amundsen      | Created type [Log Data]
```

... and:
image

P.S. I can confirm i see the columns still being null, as you mentioned. All in all great work! Congratz 👍

**Comment by bolkedebruin** [comment]
Correct! We found out that we need the master branch of Atlas to make the provisioned data available in admundsen.

**Comment by feng-tao** [comment]
Great work @bolkedebruin . Is the schema name "null" shown in the screenshoot also expected?

File: docker-amundsen-atlas.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+services:
+  atlas:
+      image: wbaa/rokku-dev-apache-atlas:latest
+      container_name: atlas_amundsen
+      ports:
+          - 21000:21000
+      networks:
+        - amundsennet
+      environment:
+        - ATLAS_PROVISION_EXAMPLES=true
```
Comment (feng-tao) [inline docker-amundsen-atlas.yml]: that's cool. So Atlas automatically provision example data :)

**Comment by feng-tao** [comment]
cc @jinhyukchang

**Comment by bolkedebruin** [comment]
Will do an update to this PR to switch to the release candidate of ATLAS 2.0 this should resolve the null values (it is an error in the provisioning)

File: docker-amundsen-atlas.yml (hunk 2 above)
Comment (bolkedebruin) [reply docker-amundsen-atlas.yml]: Yes no extra work required ;-)

**Pull request merged**